src/ui/view/dataverse/component/generalMetadata/generalMetadataValue.tsx

```diff
@@ -0,0 +1,51 @@
+import { useTranslation } from 'react-i18next'
+import * as E from 'fp-ts/Either'
+import { pipe } from 'fp-ts/lib/function'
+import { fromPredicate } from 'fp-ts/lib/Either'
+import type { ItemGeneralMetadata } from '@/ui/view/dataverse/types'
+import './generalMetadata.scss'
+import './i18n/index'
+import { DateFromTo } from './dateFromTo'
+
+const isPeriodGeneralMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is ItemGeneralMetadata & { property: 'period'; value: [string, string] } =>
+  metadata.property === 'period'
+
+const isGeneralMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is ItemGeneralMetadata & { value: string } => metadata.property !== 'period'
+
+type MetadataValueProps = {
+  metadata: ItemGeneralMetadata
+}
+
+export const GeneralMetadataValue = ({ metadata }: MetadataValueProps): JSX.Element | null => {
+  const namespace = 'generalMetadata'
+  const { t, i18n } = useTranslation(namespace)
+  const { value, property } = metadata
+  if (isGeneralMetadata(metadata)) {
+    const translationKeyExists = i18n.exists(`${namespace}.topic.value.${value}`, {
+      ns: namespace,
+      lng: i18n.language
+    })
+    const translatedValue = pipe(
+      value,
+      fromPredicate(
+        () => translationKeyExists,
+        () => value
+      ),
+      E.map(val => t(`${namespace}.${property}.value.${val}`)),
+      E.getOrElse(() => value)
+    )
+    return <span>{translatedValue}</span>
+  }
+
+  if (isPeriodGeneralMetadata(metadata)) {
+    const [startDateValue, endDateValue] = metadata.value
+    return <DateFromTo fromDateString={endDateValue} toDateString={startDateValue} />
+  }
+
+  console.warn('Metadata is not of type ItemGeneralMetadata')
```

Comment:
I'm not a big fan of logging (is it drop in prod?).
But if you want to log during development to help developers quickly identify potential problems, you should provide more context (context is king).  In this case, the message isn't very helpful. Why not:

```suggestion
  console.warn('Metadata is not of type %s: %O', 'ItemGeneralMetadata', metadata)
```

Reply:
@ccamel  Yes it is currently printed in the console.
We can either:
- display it on in dev based on `NODE_ENV`.
- return a component that display something an explicit error instead of `null`.
We can even do both.
What would you suggest? Thanks

Reply:
AFAI there's a vitejs plugin for that: https://www.npmjs.com/package/vite-plugin-remove-console.
But don't manage it here, create a new ticket for that. We will address it later. Thx :pray: 

Just improve the logs to provide more context.

Reply:
Sounds a better way to handle console logging!